lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -513,12 +513,12 @@ public static Domain getDomain(
             return Domain.all(type);
         }
 
-        long totalNullCount = nullCounts.stream()
+        Optional<Long> totalNullCount = nullCounts.map(pageNullCounts -> pageNullCounts.stream()
```

Comment:
This is a little hard to read with the nested map()/stream()/mapToLong().

Also, `totalNullCount` is not used as an Optional anywhere else (the value is read unconditionally), so you might as well just do:

```java
long totalNullCount = 0;
if (nullCounts.isPresent()) {
    totalNullCount = nullCounts.orElseThrow().stream()
                    .mapToLong(value -> value)
                    .sum();
}